hivemq-edge/src/main/java/com/hivemq/persistence/util/FutureUtils.java

```diff
@@ -76,4 +79,14 @@ public void onFailure(final Throwable throwable) {
             }
         }, MoreExecutors.directExecutor());
     }
+
+    public static void syncFuture(final @NotNull Future<?> future) {
+        try {
+            future.get();
+        } catch (final InterruptedException e) {
+            Thread.currentThread().interrupt();
+        } catch (final ExecutionException e) {
```

Comment:
Shouldn't the caller know about the failure?

Reply:
yeah method is not used yet. but i had the same thought. not sure